### ISSUE TITLE
enforce-switch-style: false positive with pure return based switch

### ISSUE THREAD

**Issue by ccoVeille** [mgechev/revive#1438]
**Describe the bug**

**To Reproduce**
1. I updated revive `go install github.com/mgechev/revive@latest`
2. I run it with the following flags & configuration file:

```
type Severity int

const (
	Error Severity = iota
	Warning
	Informational
)

func (s Severity) String() string {
	switch s { // FALSE POSITIVE got: switch must have a default case clause
	case Error:
		return "err"
	case Warning:
		return "warning"
	case Informational:
		return "info"
	}
	return "unknown"
}
```

**Expected behavior**

Having no error reported

**Additional context**

Right now, it wants me to write this, which I dislike.

```
func (s Severity) String() string {
	switch s { 
	case Error:
		return "err"
	case Warning:
		return "warning"
	case Informational:
		return "info"
	default:
		return "unknown"
	}
}
```

**Comment by ccoVeille** [comment]
@chavacava I thought about what you said [here](https://github.com/mgechev/revive/pull/1436#issue-3259857949) for `enforce-else` rule idea you described

> The rule will not report if ... else if chains where all the (then) branches end with return or break

I feel like the switch should do the same.

While the example reported in the issue description is very basic, and should not trigger an error according to me.

Please note, here I chose to provide an `enum style` switch, and not something like an `expression-only` switch, that is still yet to be discussed separately in https://github.com/mgechev/revive/issues/1434

**Comment by chavacava** [comment]
@ccoVeille exactly.
I ll take this fix

**Comment by gannett-ggreer** [comment]
This makes me think of the https://github.com/nishanths/exhaustive linter although it has a slightly different purpose.

**Comment by ccoVeille** [comment]
> This makes me think of the [nishanths/exhaustive](https://github.com/nishanths/exhaustive) linter although it has a slightly different purpose.

Yes, it leads to confusion. Linters are overlapping. It's normal, revive is a "meta linter", some rules are already in other linters.
It's OK because revive can be used separately than golangci-lint

But, in fact, the more I try the `enforce-switch-style` rule on my code, the more there is a problem with the current implementation of the rule.
Also, if we consider #1434 there is another problem.
**For me, something about enforcing the position of the default case should not care if the default case is present.**

It's what exhaustive does well, and it's something that revive could do.
but I feel like the detection of "the default is missing in the switch" should be done in another revive rule.

Said otherwise:
- revive can do something that exhaustive linter does
- revive implementation needs to be improved as stated in the initial description of this issue, and here https://github.com/mgechev/revive/issues/1434
- revive should move this detection in a dedicated rule
- enforce-switch-style should be only about the detection of the position of the "default" case.

@chavacava @denisvmedia @alexandear

**Comment by ccoVeille** [comment]
Or the current implementation of the rule should not check for the existence of default case by default.

Please note, no matter how it would be structured. The issue initially reported in this issue description remains, also #1434 still needs to be considered.

**Comment by chavacava** [comment]
@alexandear are you working on this issue ?

**Comment by alexandear** [comment]
> @alexandear are you working on this issue ?

No, I'm not.
